lib/cloud/file-storage/awsStorage.js

```diff
@@ -0,0 +1,91 @@
+var q = require('q');
+var s3 = require('s3');
+var path = require('path');
+
+var config;
+var awsClient;
+
+function init(storageConfiguration) {
+  config = storageConfiguration;
+  validateConfig();
+  awsClient = s3.createClient(config);
+}
+
+function writeFile(namespace, fileName, fileLocation) {
+  var file;
+  if (namespace) {
+    file = path.join(namespace, fileName);
+  } else {
+    file = fileName;
+  }
+  
+  var params = {
+    localFile: fileLocation,
+    ACL: 'authenticated-read',
+    s3Params: {
+      Bucket: config.bucket,
+      Key: file
+    }
+  };
+  
+  var deferred = q.defer();
```

Comment:
We need to migrate to bluebird, but possibly using similar patterns across the all modules.
Let's address this in our backlog as there are multiple migration steps and we should be consistent.